cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -167,44 +183,53 @@ int64_t WholeStageResultIterator::runtimeMetric(
-  got = confMap_.find(kVeloxMemoryCap);
-  if (got != confMap_.end()) {
-    try {
-      auto maxMemory = (long)(std::stol(got->second));
-      configs[velox::core::QueryConfig::kMaxPartialAggregationMemory] = std::to_string(maxMemory);
-    } catch (const std::invalid_argument&) {
-      throw std::runtime_error("Invalid off-heap memory size.");
-    }
+  try {
+    // Set the max memory of partial aggregation as 3/4 of offheap size.
+    auto maxMemory =
+        (long)(0.75 * (double)std::stol(getConfigValue(kSparkOffHeapMemory, std::to_string(facebook::velox::memory::kMaxMemory))));
+    configs[velox::core::QueryConfig::kMaxPartialAggregationMemory] = std::to_string(maxMemory);
+  } catch (const std::invalid_argument&) {
+    throw std::runtime_error("Invalid off-heap memory size.");
   }
   // To align with Spark's behavior, set casting to int to be truncating.
   configs[velox::core::QueryConfig::kCastIntByTruncate] = std::to_string(true);
-  configs[velox::core::QueryConfig::kSpillEnabled] = std::to_string(true);
-  configs[velox::core::QueryConfig::kAggregationSpillEnabled] = std::to_string(true);
-  configs[velox::core::QueryConfig::kJoinSpillEnabled] = std::to_string(true);
-  configs[velox::core::QueryConfig::kOrderBySpillEnabled] = std::to_string(true);
+
+  // Spill configs
+  configs[velox::core::QueryConfig::kSpillEnabled] = getConfigValue(kSpillEnabled, "true");
+  configs[velox::core::QueryConfig::kAggregationSpillEnabled] = getConfigValue(kAggregationSpillEnabled, "true");
+  configs[velox::core::QueryConfig::kJoinSpillEnabled] = getConfigValue(kJoinSpillEnabled, "true");
+  configs[velox::core::QueryConfig::kOrderBySpillEnabled] = getConfigValue(kOrderBySpillEnabled, "true");
   configs[velox::core::QueryConfig::kAggregationSpillMemoryThreshold] =
-      std::to_string(0); // spill only when input doesn't fit
-  configs[velox::core::QueryConfig::kJoinSpillMemoryThreshold] = std::to_string(0); // spill only when input doesn't fit
+      getConfigValue(kAggregationSpillMemoryThreshold, "0"); // spill only when input doesn't fit
+  configs[velox::core::QueryConfig::kJoinSpillMemoryThreshold] =
+      getConfigValue(kJoinSpillMemoryThreshold, "0"); // spill only when input doesn't fit
   configs[velox::core::QueryConfig::kOrderBySpillMemoryThreshold] =
-      std::to_string(0); // spill only when input doesn't fit
+      getConfigValue(kOrderBySpillMemoryThreshold, "0"); // spill only when input doesn't fit
   queryCtx->setConfigOverridesUnsafe(std::move(configs));
 }
 
 std::shared_ptr<velox::Config> WholeStageResultIterator::createConnectorConfig() {
   std::unordered_map<std::string, std::string> configs = {};
-  auto got = confMap_.find(kCaseSensitive);
-  if (got != confMap_.end()) {
-    configs[velox::connector::hive::HiveConfig::kCaseSensitive] = got->second;
-  }
+  configs[velox::connector::hive::HiveConfig::kCaseSensitive] = getConfigValue(kCaseSensitive, "true");
```

Comment:
The spark default config value is `false`

Reply:
I would suggest trying to make it `false` in a separate topic / PR. In Velox it's defaulted to `true`. So this change is subject to the previous logic here.